src/psyclone/psyGen.py

```diff
@@ -6265,6 +6385,28 @@ def __str__(self):
             result += str(entity)
         return result
 
+    def reference_accesses(self, var_accesses):
+        '''Get all variable access information. All variables used as indices
+        in the access of the array will be added as READ.
+        :param var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+        '''
+
+        # This will set the array-name as READ
+        super(Array, self).reference_accesses(var_accesses)
+
+        # Now add all children: Note that the class Reference
+        # does not recurse to the children, so at this stage no
+        # index information has been stored:
+        list_indices = []
+        for child in self._children:
+            child.reference_accesses(var_accesses)
+            list_indices.append(child)
+
+        if list_indices:
+            var_info = var_accesses.get_varinfo(self._reference)
+            var_info.get_all_accesses()[0].set_indices(list_indices)
```

Comment:
I'm not comfortable with the fact that the first entry in the list of accesses is "special" and is the only one with information that the variable is an array. Ideally I think we'd solve this with a link to the appropriate entry in the SymbolTable but we don't yet use that in the NEMO API. Maybe `var_info` should just have a `set_indices` method which would then hide the fact that we're storing it in element 0 of the list?

Reply:
Agreed - that line was actually wrong, the 0 must be a -1 to modify the last element that was just added before. I've fixed it and added a comment to describe this.